apps/meteor/app/livechat/server/lib/contacts/ContactMerger.ts

```diff
@@ -229,6 +236,15 @@ export class ContactMerger {
 			}
 		}
 
+		if (newActivities.length) {
+			const newActivity = newActivities.shift();
+			if (newActivity) {
+				const distinctActivities = new Set([...newActivity, ...(contact.activity || [])]);
+				const latestActivities = Array.from(distinctActivities).sort().slice(-12);
```

Comment:
what is -12? a comment maybe (I am not opinionated on naming everything but at least a comment to remove confusion 🙈)

Reply:
also, sort on a string list, named as latestActivites, I am confused slightly. How is a string sort fetching conceptually 'latest' activities?

Reply:
got this from a test
```
const period = moment().format('YYYY-MM');
```

so the strings are effectively dates and 12 my guess is months?

could we type `activity` instead of `string[]` more like
```
type YearMonthString = string;
{ activity: YearMonthString[] }
```
?

Reply:
I wanted to keep the "rule" we had [here](https://github.com/RocketChat/Rocket.Chat/blob/0300995515f6f22a62cb57f3a584d054ff00e67b/apps/meteor/server/models/raw/LivechatVisitors.ts#L472). When inserting new month activities, we always keep only the latest 12 ones
I'll keep the type as string, but use a new name to make this clear as you suggested

Reply:
Yep, just the last 12 are saved as activity. Got curious, can't we do this on mongo? the slicing?

Reply:
> Got curious, can't we do this on mongo? the slicing?
We could, but slice can't be performed inside a `$set`. So we'd need an extra db call to perform a push with a slice instead -- so I thought it wasn't worth it since I can do the slice myself and set it with a single DB call

Reply:
Yeah but $set and $push can coexist on the same update statement 👀 

the only reason i think justifies is the sortin 🤔 so I'll let you decide if there's a way to use push or keep what you doing here.

Reply:
> the only reason i think justifies is the sortin 🤔 so I'll let you decide if there's a way to use push or keep what you doing here.

That's it, I was afraid of using push because it could mess up the order :thinking: we call this function once per similar contact that should be merged, so we can't just assume that the new activities we're inserting are the latest (and make previous elements be sliced out of the array)
I'll move on with `$set`, then